src/components/Sidebar.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import SwipeableDrawer from '@bit/mui-org.material-ui.swipeable-drawer'
+import { useSelector, useDispatch } from 'react-redux'
+import { isMobile } from '../browser'
+import sortBy from 'lodash.sortby'
+import reverse from 'lodash.reverse'
+
+/** This component returns ellipsized path such as A • B • ... • E
+ *
+ * thoughtsLimit is the max number of thoughts to be shown in path without ellipsized
+ * for A/B/C/D/E/F
+ * if thoughtsLimit=3 => A • B • ... • F
+ * if thoughtsLimit=5 => A • B • C • D • ... • F
+ * if thoughtsLimit=6 => A • B • C • D • E • F
+ *
+ * charLimit is the max number of character of a thought value that can be shown , if maxed out will be replaced by ..
+ *
+ *
+ * if charLimit=7 and thoughtsLimit=5
+ * path: To-do/Read/Encryption/keys/private/hash
+ * it will give To-do • Read • Encrypti..•keys • ... • hash
+*/
```

Comment:
After some additional testing, I realize that we need to go with the original vision of breadcrumbs as separate links, as implemented in the current `Breadcrumbs` component. I tried out your implementation of a single link for the whole path, but from a user experience perspective it was falling short. Sorry I didn't catch this earlier. Thank you for understanding. 

If you'll notice in the current `Breadcrumbs` component, each thought in the given context/path is individually clickable. There is no downside to this behavior and it gives the user more control, allowing them to navigate to ancestors with fewer clicks.

I think the best approach would be to add the ellipsis logic to the existing `Breadcrumbs` component. When an ellipsis is tapped, it should expand into the full thought/path. We should be able to track the ellipsis expansion with a state hook since it does not need to be persisted.

Otherwise your ellipsis logic is great and I appreciate how configurable it is.